### PR TITLE
Add reader for hamamatsu .img format

### PR DESCRIPTION
**Pull request by pietsjoh**
Add support for reading Hamamatsu's .img file format.

### Progress of the PR
- [x] basic parsing of the file
- [x] map metadata
- [x] add tests,
- [x] update docstring (if appropriate),
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/rosettasciio/blob/main/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] cover different acquisition modes with testfiles
- [x] reproduce plots
- [x] update tests after problem fix
- [x] ready for review.

### Minimal example of the bug fix or the new feature
```python
from rsciio.hamatsu import file_reader
file_reader("file.img", reader="Hamamatsu")
```

### CODE REVIEW

**Comment by jlaehne** [comment]
@Attolight-NTappy and @gkusch - I know you are happy using the tiff import of Hamamatsu Streak images, but maybe you could also have a look at this reader for the Hamamatsu format?

**Comment by gkusch** [comment]
I will try to export a Streak in the Format next time I am at the microscope and compare with .tiff Currently measurements done with the Attolight Software controlling the Streak will be autosaved as .tiff so not super relevant for us in general. Are there any inherent advantages of this format vs tif or is it more a matter of providing choice?

**Comment by jlaehne** [comment]
> I will try to export a Streak in the Format next time I am at the microscope and compare with .tiff Currently measurements done with the Attolight Software controlling the Streak will be autosaved as .tiff so not super relevant for us in general. Are there any inherent advantages of this format vs tif or is it more a matter of providing choice?

According to the Hamamatsu specs, `img` supports 32 bit data, while their `tif` export supports only 16 bit data.
See https://github.com/bigphatjan/itex2ascii/blob/master/HiPicFileFormat.pdf

Both readers do read in the `original_metadata`. The one for `img` already maps some information to `metadata`.

But I'm curious what your comparison would give.

**Comment by pietsjoh** [comment]
> While reproducing plots, I discovered some problems. I didn't put much thought into reading the data block, because I thought it was just a block of continuous data. Hopefully, changing the endianness fixes the problem. But the plot still slightly differs compared to the original from a matlab script (which performs multiple steps). I will further check this and try to reproduce it exactly.

The problem is fixed now.

**Comment by jlaehne** [comment]
Indeed, the data looks good now. I compared a real dataset for which I have both the `.img` and `.tif` files.

**Comment by jlaehne** [comment]
> According to the Hamamatsu specs, `img` supports 32 bit data, while their `tif` export supports only 16 bit data.
> See https://github.com/bigphatjan/itex2ascii/blob/master/HiPicFileFormat.pdf

We should include this format documentation in the documentation under `File Specifications` for later reference.

File: rsciio/hamamatsu/specifications.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name: 'Hamamatsu'
```
[the remark this answers is not in the document]
Comment (jlaehne) [reply rsciio/hamamatsu/specifications.yaml:1]: Lets stick with the name, as no one came up with a better name. Other Hamamatsu formats could be added to the same reader, if needed.

File: docs/supported_formats/hamamatsu.rst [hunk 2]
```diff
@@ -0,0 +1,21 @@
+.. _hamamatsu-format:
```
[the remark this answers is not in the document]
Comment (pietsjoh) [reply docs/supported_formats/hamamatsu.rst:1]: It is added in `docs/supported_formats/index.rst`.

File: rsciio/hamamatsu/_api.py [hunk 3]
```diff
@@ -0,0 +1,529 @@
...
+
+
+def file_reader(filename, lazy=False, use_uniform_signal_axes=False, **kwds):
+    """Reads Hamamatsu's ``.img`` file.
+
+    Parameters
+    ----------
+    %s
+    %s
+
```
Comment (jlaehne) [inline rsciio/hamamatsu/_api.py:505]: Add `use_uniform_signal_axis` parameter, e.g. by copying the docstring from jobin-yvon (or by adding it to `_docstrings`)

Comment (pietsjoh) [reply rsciio/hamamatsu/_api.py:510]: I copied it for now and will do a follow up PR, where I add this to _docstrings and move the signal axis fit to utils.

**Review by jlaehne — APPROVED** [review]

**Pull request merged**